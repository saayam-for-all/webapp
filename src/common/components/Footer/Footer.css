/* ---------------------------------------------------
   1) Base & inner wrapper
--------------------------------------------------- */
.footer {
  background-color: #333;
  padding: 24px 20px;
  font-family: sans-serif;
  color: #fff;
}

.footer-inner {
  max-width: 1200px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  gap: 20px;
}

/* ---------------------------------------------------
     2) Top row + single divider wrapper
  --------------------------------------------------- */
.footer-top-wrapper {
  display: inline-block; /* shrinkâ€‘wrap around its contents */
  position: relative; /* for absolute <hr> positioning */
  margin: 0 auto; /* center it in the footer */
  padding-bottom: 20px; /* space between divider & bottom grid */
}

.footer-top {
  display: grid;
  grid-template-columns: auto 1fr auto;
  align-items: center;
  gap: 20px;
}

.footer-logo img {
  width: 50px;
  height: 50px;
  object-fit: cover;
  border-radius: 50%;
}

.footer-nav {
  display: flex;
  justify-content: center;
  gap: 40px;
  flex-wrap: wrap;
  justify-self: center;
}

.footer-nav a {
  color: #fff;
  text-decoration: none;
  font-size: 13px;
  white-space: nowrap;
}

.footer-nav a:hover {
  text-decoration: underline;
}

.footer-contact button {
  background-color: #2a72ff;
  color: #fff;
  border: none;
  padding: 12px 24px;
  border-radius: 30px;
  font-size: 14px;
  font-weight: bold;
  cursor: pointer;
  white-space: nowrap;
  transition: background-color 0.2s;
}

.footer-contact button:hover {
  background-color: #1e5dcc;
}

/* Divider positioned at the bottom of the wrapper */
.footer-divider {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  max-width: 100%;
  margin: 0;
  border: none;
  border-top: 1px solid rgba(255, 255, 255, 0.15);
}

/* ---------------------------------------------------
     3) Bottom grid: 4 equal columns
  --------------------------------------------------- */
.footer-bottom-grid {
  display: grid;
  grid-template-columns: auto repeat(3, auto);
  gap: 20px;
  max-width: 600px; /* match your divider width */
  margin: 0 auto;
  align-items: center;
}

.footer-copy {
  text-align: left;
  font-size: 13px;
  color: #ccc;
  white-space: nowrap;
}

.footer-link-cell {
  text-align: center;
}

.footer-link-cell a {
  color: #ccc;
  text-decoration: none;
  font-size: 13px;
}

.footer-link-cell a:hover {
  text-decoration: underline;
}

/* ---------------------------------------------------
     4) Mobile responsive
  --------------------------------------------------- */
@media (max-width: 768px) {
  .footer-top {
    grid-template-columns: 1fr;
    text-align: center;
  }
  .footer-contact {
    justify-self: center;
  }

  .footer-bottom-grid {
    grid-template-columns: 1fr;
    text-align: center;
    gap: 10px;
  }
}
