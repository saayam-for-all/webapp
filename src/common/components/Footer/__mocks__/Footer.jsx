export default function Footer() {
  return <mock-footer />;
}
