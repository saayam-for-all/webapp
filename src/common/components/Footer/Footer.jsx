import React from "react";
import "./Footer.css";
import logo from "../../../assets/logo.svg";

export default function Footer() {
  return (
    <footer className="footer">
      <div className="footer-inner">
        {/* 1) Wrap top row + divider */}
        <div className="footer-top-wrapper">
          {/* Top row: logo | nav | contact */}
          <div className="footer-top">
            <div className="footer-logo">
              <img src={logo} alt="Saayam logo" />
            </div>

            <nav className="footer-nav" aria-label="Footer navigation">
              <a href="/">Home</a>
              <a href="/our-mission">About Us</a>
              <a href="/">Mission &amp; Vision</a>
              <a href="/directors">Our Team</a>
              <a href="/">What We Offer</a>
              <a href="/donate">Donate</a>
            </nav>

            <div className="footer-contact">
              <a href="/contact">
                <button type="button">Contact Us</button>
              </a>
            </div>
          </div>
          {/* Divider exactly under that row */}
          <hr className="footer-divider" />
        </div>
        {/* /.footer-top-wrapper */}

        {/* 2) Bottom grid: copyright + links */}
        <div className="footer-bottom-grid">
          <div className="footer-copy">
            Copyright © 2025 Saayam For All. All rights reserved.
          </div>
          <div className="footer-links">
            <a href="/sitemap">Site Map</a>
            <a href="#">Terms and Conditions</a>
            <a href="#">Privacy Policy</a>
          </div>
        </div>
        {/* /.footer-bottom-grid */}
      </div>
      {/* /.footer-inner */}
    </footer>
  );
}
