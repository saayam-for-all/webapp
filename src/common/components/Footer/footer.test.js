import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Footer from "./Footer.jsx";
import React from "react";

test("contains copyright info", () => {
  render(<Footer />);
  const copyrightMessage = "Copyright Â© 2024 Saayam. All Rights Reserved";
  expect(screen.getByText(copyrightMessage)).toBeInTheDocument();
});
