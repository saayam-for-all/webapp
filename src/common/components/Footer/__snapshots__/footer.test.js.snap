// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer"
      >
        <div
          class="footer-inner"
        >
          <div
            class="footer-top-wrapper"
          >
            <div
              class="footer-top"
            >
              <div
                class="footer-logo"
              >
                <img
                  alt="Saayam logo"
                  src=""
                />
              </div>
              <nav
                aria-label="Footer navigation"
                class="footer-nav"
              >
                <a
                  href="/"
                >
                  Home
                </a>
                <a
                  href="/our-mission"
                >
                  About Us
                </a>
                <a
                  href="/"
                >
                  Mission & Vision
                </a>
                <a
                  href="/directors"
                >
                  Our Team
                </a>
                <a
                  href="/"
                >
                  What We Offer
                </a>
                <a
                  href="/donate"
                >
                  Donate
                </a>
              </nav>
              <div
                class="footer-contact"
              >
                <a
                  href="/contact"
                >
                  <button
                    type="button"
                  >
                    Contact Us
                  </button>
                </a>
              </div>
            </div>
            <hr
              class="footer-divider"
            />
          </div>
          <div
            class="footer-bottom-grid"
          >
            <div
              class="footer-copy"
            >
              Copyright © 2025 Saayam For All. All rights reserved.
            </div>
            <div
              class="footer-links"
            >
              <a
                href="/sitemap"
              >
                Site Map
              </a>
              <a
                href="#"
              >
                Terms and Conditions
              </a>
              <a
                href="#"
              >
                Privacy Policy
              </a>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer"
    >
      <div
        class="footer-inner"
      >
        <div
          class="footer-top-wrapper"
        >
          <div
            class="footer-top"
          >
            <div
              class="footer-logo"
            >
              <img
                alt="Saayam logo"
                src=""
              />
            </div>
            <nav
              aria-label="Footer navigation"
              class="footer-nav"
            >
              <a
                href="/"
              >
                Home
              </a>
              <a
                href="/our-mission"
              >
                About Us
              </a>
              <a
                href="/"
              >
                Mission & Vision
              </a>
              <a
                href="/directors"
              >
                Our Team
              </a>
              <a
                href="/"
              >
                What We Offer
              </a>
              <a
                href="/donate"
              >
                Donate
              </a>
            </nav>
            <div
              class="footer-contact"
            >
              <a
                href="/contact"
              >
                <button
                  type="button"
                >
                  Contact Us
                </button>
              </a>
            </div>
          </div>
          <hr
            class="footer-divider"
          />
        </div>
        <div
          class="footer-bottom-grid"
        >
          <div
            class="footer-copy"
          >
            Copyright © 2025 Saayam For All. All rights reserved.
          </div>
          <div
            class="footer-links"
          >
            <a
              href="/sitemap"
            >
              Site Map
            </a>
            <a
              href="#"
            >
              Terms and Conditions
            </a>
            <a
              href="#"
            >
              Privacy Policy
            </a>
          </div>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
